clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java

```diff
@@ -132,7 +131,7 @@ public AbstractResponse getErrorResponse(Throwable e) {
         Map<TopicPartition, ProduceResponse.PartitionResponse> responseMap = new HashMap<>();
 
         for (Map.Entry<TopicPartition, MemoryRecords> entry : partitionRecords.entrySet()) {
-            responseMap.put(entry.getKey(), new ProduceResponse.PartitionResponse(Errors.forException(e).code(), ProduceResponse.INVALID_OFFSET, Record.NO_TIMESTAMP));
+            responseMap.put(entry.getKey(), new ProduceResponse.PartitionResponse(Errors.forException(e)));
```

Comment:
nit: seems there's no need to create a new `PartitionResponse` for each partition?